src/MySqlConnector/Authentication/IAuthenticationPlugin.cs

```diff
@@ -19,4 +19,14 @@ public interface IAuthenticationPlugin
 	/// Method Switch Request Packet</a>.</param>
 	/// <returns>The authentication response.</returns>
 	byte[] CreateResponse(string password, ReadOnlySpan<byte> authenticationData);
+
+	/// <summary>
+	/// create password hash for fingerprint verification
+	/// </summary>
+	/// <param name="password">The client's password.</param>
+	/// <param name="authenticationData">The authentication data supplied by the server; this is the <code>auth method data</code>
+	/// from the <a href="https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::AuthSwitchRequest">Authentication
+	/// Method Switch Request Packet</a>.</param>
+	/// <returns>password hash</returns>
+	byte[] CreatePasswordHash(string password, ReadOnlySpan<byte> authenticationData);
```

Comment:
Adding an interface member is a breaking change; we probably need to introduce a new interface to support this. 

(Not that it's very likely that anyone has ever implemented this interface...)